src/main/java/com/aws/greengrass/deployment/ConnectivityValidator.java

```diff
@@ -0,0 +1,235 @@
+        } catch (ExecutionException e) {
+            message = "Mqtt client validation completed exceptionally: " + e.getMessage();
+            logger.atError().cause(e).log(message + withNewConfigMessageSuffix);
+        } catch (InterruptedException e) {
+            message = "Mqtt client connection was interrupted: " + e.getMessage();
+            logger.atInfo().cause(e).log(message);
+            Thread.currentThread().interrupt();
+        } finally {
+            if (connection != null) {
+                connection.disconnect();
+                connection.close();
+            }
+        }
+
+        totallyCompleteFuture
+                .complete(new DeploymentResult(DeploymentResult.DeploymentStatus.FAILED_NO_STATE_CHANGE,
+                        new ComponentConfigurationValidationException(message,
+                                DeploymentErrorCode.NUCLEUS_CONNECTIVITY_CONFIG_NOT_VALID)));
+        return false;
+    }
+
+    /**
+     * Creates an HTTP client and checks if the device can connect to AWS.
+     *
+     * @param totallyCompleteFuture Future will be updated if validation fails
+     */
+    @SuppressWarnings("PMD.AvoidCatchingGenericException")
+    public boolean httpClientCanConnect(CompletableFuture<DeploymentResult> totallyCompleteFuture) {
+        logger.atDebug().log("Checking HTTP client can connect");
+
+        String message;
+        Pair<SdkHttpClient, GreengrassV2DataClient> pair = factory.createClientFromConfig(deploymentConfiguration);
+        try (SdkHttpClient httpClient = pair.getLeft();
+             GreengrassV2DataClient greengrassV2DataClient = pair.getRight()) {
+            ListThingGroupsForCoreDeviceRequest request = ListThingGroupsForCoreDeviceRequest.builder()
+                    .coreDeviceThingName(Coerce.toString(deploymentConfiguration.getThingName())).build();
+            greengrassV2DataClient.listThingGroupsForCoreDevice(request);
+            return true;
+        } catch (Exception e) {
+            message = "HTTP client validation failed due to: " + Utils.generateFailureMessage(e);
+            logger.atError().cause(e).log(message + withNewConfigMessageSuffix);
+        }
+
+        totallyCompleteFuture
+                .complete(new DeploymentResult(DeploymentResult.DeploymentStatus.FAILED_NO_STATE_CHANGE,
+                        new ComponentConfigurationValidationException(message,
+                                DeploymentErrorCode.NUCLEUS_CONNECTIVITY_CONFIG_NOT_VALID)));
+        return false;
+    }
+
+    /**
+     * Perform connectivity validation if configs have changed meaningfully.
+     *
+     * @param totallyCompleteFuture      Future will be updated if validation fails
+     * @param currentDeviceConfiguration current configs the device is using
+     */
+    @SuppressWarnings("PMD.SimplifyBooleanReturns")
+    public boolean validate(CompletableFuture<DeploymentResult> totallyCompleteFuture,
```

Comment:
I think you should explain the different stages of validation in the method doc

Reply:
What do you think about this description:

```
Perform connectivity validation if configs have changed meaningfully.
Checks if validation is enabled
If MQTT related configurations have been changed validate MQTT connectivity
Then if HTPP related configurations have been changed validate HTTP connectivity
```

Reply:
yep, that looks good